extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -120,59 +105,14 @@ public function index() {
 	 * Accesses top wiki articles for right rail, see PLA-466
 	 */
 	public function topWikiArticles() {
-		global $wgLang;
-		$pages = [];
-		try {
-			$pageData = $this->app->sendRequest( 'ArticlesApiController', 'getTop', [ 'namespaces' => 0 ] )->getData();
-			$ids = [];
-			$counter = 0;
-			foreach ( $pageData['items'] as $pageDatum ) {
-				$ids[] = $pageDatum['id'];
-				if ( $counter++ >= 12 ) {
-					break;
-				}
-			}
-			if ( ! empty( $ids ) ) {
-				$params = [ 'ids' => implode( ',', $ids ), 'height' => 80, 'width' => 80, 'abstract' => 120 ];
-				$detailResponse = $this->app->sendRequest( 'ArticlesApiController', 'getDetails', $params )->getData();
-				$dimensions = $this->getHotArticleImageDimensions();
-				foreach ( $detailResponse['items'] as $id => $item ) {
-					if ( ! empty( $item['thumbnail'] ) ) {
-						$item['thumbnailSize'] = "small";
-						//get the first one image from imageServing as it needs other size
-						if ( empty( $pages ) ) {
-							$is = new ImageServing( [ $id ], $dimensions['width'], $dimensions['height'] );
-							$result = $is->getImages( 1 );
-							if ( ! empty( $result[ $id ][ 0 ][ 'url' ] ) ) {
-								$item[ 'thumbnail' ] = $result[ $id ][ 0 ][ 'url' ];
-								$item['thumbnailSize'] = "large";
-							}
-						}
-						//render date
-						$item[ 'date' ] = $wgLang->date( $item[ 'revision' ][ 'timestamp' ] );
-						$item = $this->processArticleItem( $item, 100 );
-						$pages[] = $item;
-					}
-				}
-			}
-		} catch ( Exception $e ) { } // ignoring API exceptions for gracefulness
-		$this->setVal( 'pages', $pages );
+		$this->response->setVal( 'pages', $this->getVal( 'pages', [] ) );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "pages" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)